packages/preview-middleware-client/src/cpe/facade.ts

```diff
@@ -0,0 +1,95 @@
+import type { IconDetails } from '@sap-ux-private/control-property-editor-common';
+import type { UI5Facade } from './types';
+import Component from 'sap/ui/core/Component';
+import type Element from 'sap/ui/core/Element';
+import type ElementOverlay from 'sap/ui/dt/ElementOverlay';
+import type { ID } from 'sap/ui/core/library';
+import OverlayRegistry from 'sap/ui/dt/OverlayRegistry';
+import OverlayUtil from 'sap/ui/dt/OverlayUtil';
+import IconPool from 'sap/ui/core/IconPool';
+
+/**
+ * Gets control by id.
+ *
+ * @param id unique identifier for control
+ * @returns Element | undefined
+ */
+function getControlById<T extends Element>(id: ID): T | undefined {
+    return sap.ui.getCore().byId(id) as T;
+}
+
+/**
+ * Gets Component by id.
+ *
+ * @param id - unique identifier for control
+ * @returns Component | undefined
+ */
+function getComponent<T extends Component>(id: ID): T | undefined {
+    if (Component?.get) {
+        return Component.get(id) as T;
+    } else {
+        // Older version must be still supported until maintenance period.
+        return sap.ui.getCore().getComponent(id) as T; // NOSONAR
+    }
+}
+
+/**
+ * Gets overlay for control.
+ *
+ * @param control ui5 control
+ * @returns ElementOverlay | undefined
+ */
+function getOverlay<T extends ElementOverlay>(control: Element): T | undefined {
+    return OverlayRegistry.getOverlay(control) as T;
+}
+
+/**
+ * Gets ClosestOverlay for control.
+ *
+ * @param control ui5 control
+ * @returns ElementOverlay | undefined
+ */
+function getClosestOverlayFor<T extends ElementOverlay>(control: Element): T | undefined {
+    return OverlayUtil.getClosestOverlayFor(control) as T;
+}
+
+/**
+ * Get ui5 icons.
+ *
+ * @returns IconDetails[]
+ */
+function getIcons(): IconDetails[] {
+    return IconPool.getIconNames('undefined')
+        .map((icon: string) => {
+            const iconInfo = IconPool.getIconInfo(icon) as IconDetails;
+            return {
+                name: icon.toLowerCase(),
+                content: iconInfo.content,
+                fontFamily: iconInfo.fontFamily
+            } as IconDetails;
+        })
+        .sort((item1: IconDetails, item2: IconDetails) => {
+            if (item1.name < item2.name) {
+                return -1;
+            }
+            if (item1.name > item2.name) {
+                return 1;
+            }
+            return 0;
+        });
+}
+
+/**
+ * A facade for UI5 framework methods.
+ *
+ * @returns UI5Facade
+ */
+export function createUi5Facade(): UI5Facade {
```

Comment:
Why do you export a function that dynamically creates the facade? Why wouldn't you just export the functions directly?

Reply:
Facade was was created, because we didn't have a good way how to deal with UI5 dependencies before and to make testing easier. However, I think for both cases we now have better solutions, so I think we probably can remove some of these wrappers all together.

Reply:
Fixed by [bf2ba99](https://github.com/SAP/open-ux-tools/pull/1197/commits/bf2ba99323e742ede700424cd3881a97d7633287).

Reply:
added it as follow up to #1328